### ISSUE TITLE
vncserver: socket failed: Permissions denied Error

### ISSUE THREAD

**Issue by TunahanYeniyayla** [meefik/linuxdeploy#1160]
When I press the start button, I get the error in the header (vncserver: socket failed: permission denied). All my settings are in the attached pictures. I can use ssh. I get the same error even though I try in other settings and linux.

[https://i.hizliresim.com/ZngapG.jpg](url)
[https://i.hizliresim.com/gbzDp3.png](url)
[https://i.hizliresim.com/l428pJ.png](url)

**Comment by Lewinator56** [comment]
I've been having this issue lately too, only seems to have started happening after I deleted and re-installed Linux. From what I can tell, it's an issue with the implementation of 'chroot_exec', if you switch on the debug log in settings you will get this error:
`-su: 2: [: x: unexpected operator`
I've tried changing the code in the deploy file for the VNC server but no hope.

I'm also on a Xiaomi device, so wondering if it's device specific, though the issue seems to be in how magisk is passing the chroot_exec command to the container. So it might be a magisk issue.

**Comment by TunahanYeniyayla** [comment]
I deleted all data on twrp except storage. I installed the phone as if I was installing from scratch. I got the same error again, even though I started all over again. It seems to me that it originates from xiaomi. But I don't know how we can solve it. I was using it before.

**Comment by Lewinator56** [comment]
im not sure though, like i said, it was working perfectly fine before i reinstalled the distro, that doesn't imply any issue with xiaomi, rather with something the app is doing, something isn't being configured right, the fact it was working and now isn't means an update to the app messed with something, i haven't updated my OS since the last working install i did either

**Comment by TunahanYeniyayla** [comment]
if your problem is solved my problem will be solved. When your problem is solved, would you prepare a text and a guide for this? So what do I have to do to run the vnc again?

**Comment by Lewinator56** [comment]
yep, i will, you cant even start vnc from within the container at the moment due to a permission issue, so its definitely an issue somewhere in the deployment files in the app, ive looked through them but cant find anything, i'll keep trying though.

**Comment by Lewinator56** [comment]
got a temporary fix!
uninstall the current version of linux deploy, delete exerything to do with it and install version 2.5.0, its working properly with that, just get ubuntu installed!

**Comment by TunahanYeniyayla** [comment]
Yes !! Currently I have installed Kali and it works very well. Thank you. This means that when switching from version 2.5.0 to 2.6.0, something has been corrupted. Let's continue with this version for now.

**Comment by tichli** [comment]
Same problem for me guys. But as I am running Android 10, I cannot revert to 2.5.0 which was buggy for that Android version. I guess I'll have to wait for the next release.

**Comment by mugejie** [comment]
It is permission problem, if you ssh in, su to root, then run vncserver, it will be ok

you can look in the log file, when start up, it use the user not root to exec
    # exec vncserver
    chroot_exec -u ${USER_NAME} vncserver :${VNC_DISPLAY} -depth ${VNC_DEPTH} -dpi ${VNC_DPI} -geometry ${VNC_WIDTH}x${VNC_HEIGHT} ${VNC_ARGS}
    is_ok "fail" 

you can edit the vnc deploy.sh
or try the solution: https://forum.xda-developers.com/t/how-to-run-debian-armel-on-your-n1.622474/page-3  :

"
I get a problem when trying to run "vncserver" as non root (it works as root) "vncserver: socket failed: Permission denied". I guess its trying to bind to port 5901, so not sure what its beef is... something Android specific maybe?
hi i got the exact same problem, on how to launch vncserver as a simple user in a chrooted environment.

solved it creating a sockets group :
Create a user with adduser username. You must also create a group with numeric ID 3003 and add yourself to it, otherwise vncserver's sockets will be blocked! So,
Code:
addgroup --gid 3003 sockets
usermod -G sockets -a [I]username[/I]
I know this thread is quite old, but might help people like me who get here
"
